transformers/storage/vat/repository_test.go

```diff
@@ -1051,10 +1074,10 @@ var _ = Describe("Vat storage repository", func() {
 
 				It("deletes urn state associated with diff if it's the earliest state in the table", func() {
 					initialInk := rand.Int()
-					setupErrOne := repo.Create(diffID, headerOne.Id, urnInkMetadata, strconv.Itoa(initialInk))
+					setupErrOne := repo.Create(diffIdOne, headerOne.Id, urnInkMetadata, strconv.Itoa(initialInk))
```

Comment:
Here, with the creation of diffIdOne you've ensured the urnSnapshot is deleted right? Because before using the generic diffID hid that error?

Reply:
Yep, that's right. The key piece here is that the diff's `block_height` is the same as the `block_number` on the header that is deleted. So, we can make sure that the `urn_snapshot_block_number` that is being queried for in the `delete_obsolete_urn_snapshot` function is the correct block number's snapshot to delete.